api/nakadi-event-bus-api.yaml

```diff
@@ -827,135 +685,118 @@ definitions:
+      received_at:
+        type: string
+        description: |
+          Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of
+          the Event.
+          If set by the producer Event will be rejected.
+        format: date-time
+        example: '1996-12-19T16:39:57-08:00'
+      parent_eids:
+        type: array
+        items:
+          type: string
+          format: uuid
+          description: |
+            Event identifier of the Event that caused the generation of this Event.
+            Set by the producer.
+          example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+      flow_id:
+        description: |
+          The flow-id of the producer of this Event. As this is usually a HTTP header, this is
+          enriched from the header into the metadata by Nakadi to avoid clients having to
+          explicitly copy this.
+        type: string
+        example: 'JAh6xH4OQhCJ9PutIV_RYw'
+      partition:
+        description: |
+          Indicates the partition assigned to this Event.
 
+          Required to be set by the client if partition strategy of the EventType is
+          'user_defined'.
+        type: string
+        example: '0'
     required:
-      - metadata
+      - eid
+      - occurred_at
 
   BusinessEvent:
-    type: object
     description: |
       A Business Event.
 
-      Usually represents a status transition in a Business process.
-    allOf:
-      - $ref: '#/definitions/EventMetadata'
-    additionalProperties: true
-
-  DataChangeEventQualifier:
-    type: object
-    description: |
-      Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on
-      them.
-
-    properties:
-      data_type:
-        type: string
-        example: 'pennybags:order'
-      data_op:
-        type: string
-        enum: ['C', 'U', 'D', 'S']
-        description: |
-          The type of operation executed on the entity.
-          * C: Creation
-          * U: Update
-          * D: Deletion
-          * S: Snapshot
-    required:
-      - data_type
-      - data_op
+      Usually represents a status transition in a Business process.    
+    allOf:  
+      - $ref: '#/definitions/Event'
+      - type: object
+        properties:
+          metadata:
+              $ref: '#/definitions/EventMetadata'
+        required:
+            - metadata
 
   DataChangeEvent:
-    type: object
     description: |
       A Data change Event.
 
-      Represents a change on a resource.
+      Represents a change on a resource. Also contains indicators for the data 
+      type and the type of operation performed.
     allOf:
-      - $ref: '#/definitions/DataChangeEventQualifier'
-      - $ref: '#/definitions/EventMetadata'
-    properties:
-      data:
-        type: object
-        description: |
-            The payload of the type
-    required:
-        - data
-    additionalProperties: false
+      - $ref: '#/definitions/Event'
+      - type: object        
```

Comment:
This removes DataChangeEventQualifier and inlines that object's properties into DataChangeEvent. DataChangeEventQualifier is never going to be used outside a DataChangeEvent, plus its properties were directly inlined and not placed under a parent property (eg the way Metadata is placed under a `metadata` field).